lib/bfd.c

```diff
@@ -94,22 +94,29 @@ int bfd_validate_param(struct vty *vty, const char *dm_str, const char *rx_str,
  * bfd_set_param - Set the configured BFD paramter values
  */
 void bfd_set_param(struct bfd_info **bfd_info, uint32_t min_rx, uint32_t min_tx,
-		   uint8_t detect_mult, int defaults, int *command)
+		   uint8_t detect_mult, const char *profile, int defaults,
+		   int *command)
 {
 	if (!*bfd_info) {
 		*bfd_info = bfd_info_create();
 		*command = ZEBRA_BFD_DEST_REGISTER;
 	} else {
 		if (((*bfd_info)->required_min_rx != min_rx)
 		    || ((*bfd_info)->desired_min_tx != min_tx)
-		    || ((*bfd_info)->detect_mult != detect_mult))
+		    || ((*bfd_info)->detect_mult != detect_mult)
+		    || (profile && strcmp((*bfd_info)->profile, profile)))
 			*command = ZEBRA_BFD_DEST_UPDATE;
 	}
 
 	if (*command) {
 		(*bfd_info)->required_min_rx = min_rx;
 		(*bfd_info)->desired_min_tx = min_tx;
 		(*bfd_info)->detect_mult = detect_mult;
+		if (profile)
+			strncpy((*bfd_info)->profile, profile,
```

Comment:
`strncpy` does not guarantees that the string is NULL terminated.

Man page bit that talks about it:

```
Warning: If there is no null byte among the first n bytes of src, the string placed in dest will not be  null-terminated.
```

( https://www.man7.org/linux/man-pages/man3/strcpy.3.html )

Please, use `strlcpy` or `snprintf` instead.